src/Appwrite/Extend/Exception.php

```diff
@@ -164,6 +171,18 @@ class Exception extends \Exception
     /** Webhooks */
     public const WEBHOOK_NOT_FOUND                 = 'webhook_not_found';
 
+    /** Router */
+    public const ROUTER_UNKNOWN_HOST                = 'router_unknown_host';
+    public const ROUTER_INVALID_TYPE                = 'router_unknown_type';
+
+    /** Proxy */
+    public const RULE_CONFIGURATION_MISSING         = 'rule_configuration_missing';
+    public const RULE_RESOURCE_ID_MISSING           = 'rule_resource_id_missing';
```

Comment:
Why MISSING vs NOT_FOUND?

Reply:
It's for when you set resporceType=function, and dont provide resourceId (function ID). It's 400, not 404. It's "request validator"

Reply:
Used function not found instead.